SIPS/sip-6.md

```diff
@@ -0,0 +1,184 @@
+---
+sip: 6
+title: Deterministic Snap-specific entropy
+status: Draft
+discussions-to: https://github.com/MetaMask/SIPs/discussions/69
+author: Maarten Zuidhoorn (@Mrtenz)
+created: 2022-10-27
+---
+
+## Abstract
+
+This SIP describes a way for Snaps to get some deterministic Snap-specific entropy, based on the secret recovery phrase
+of the user and the snap ID. Since Snaps do not have access to the secret recovery phrase directly, other Snaps are
+unable to get the same entropy.
+
+Snaps can optionally specify a salt in order to generate different entropy for different use cases, for example, some
+entropy to derive new private keys from, and some other entropy to encrypt some data.
+
+The entropy can be accessed from within a Snap, using the `snap_getEntropy` JSON-RPC method.
+
+## Motivation
+
+Before this SIP, Snaps did not have a way to get some kind of deterministic entropy, without storing it on the disk. If
```

Comment:
If possible it might be interesting to use this concept for state encryption as well.